R/pkg/R/deserialize.R

```diff
@@ -197,6 +197,31 @@ readMultipleObjects <- function(inputCon) {
   data # this is a list of named lists now
 }
 
+readMultipleObjectsWithKeys <- function(inputCon) {
+  # readMultipleObjectsWithKeys will read multiple continuous objects from
+  # a DataOutputStream. There is no preceding field telling the count
+  # of the objects, so the number of objects varies, we try to read
+  # all objects in a loop until the end of the stream. The rows in
```

Comment:
comment:
Replace the statement "The rows in ..." with 
"
This function is for use by gapply. Each group of rows is followed by the grouping key for this group which is then followed by next group."